core/server/master/src/main/java/alluxio/master/file/loadmanager/LoadManager.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.loadmanager;
+
+import static alluxio.master.file.loadmanager.load.LoadInfo.LoadOptions;
+
+import alluxio.exception.AlluxioRuntimeException;
+import alluxio.exception.status.ResourceExhaustedException;
+import alluxio.grpc.LoadResponse;
+import alluxio.job.meta.JobIdGenerator;
+import alluxio.job.wire.Status;
+import alluxio.master.file.loadmanager.load.LoadInfo;
+import alluxio.master.file.loadmanager.load.BlockBatch;
+import alluxio.master.journal.Journaled;
+import alluxio.master.journal.checkpoint.CheckpointName;
+import alluxio.proto.journal.File;
+import alluxio.proto.journal.Journal;
+import alluxio.resource.CloseableIterator;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * The Load manager which controls load operations.
+ */
+public final class LoadManager implements Journaled {
+  private static final Logger LOG = LoggerFactory.getLogger(LoadManager.class);
+  private final Map<String, LoadInfo>
+          mLoadPathToInfo = Maps.newHashMap();
+  private final JobIdGenerator mJobIdGenerator = new JobIdGenerator();
+  private final Map<Long, Load> mLoads = Maps.newHashMap();
+  private final Scheduler mScheduler = new Scheduler();
+
+  /**
+   * Constructor.
+   */
+  public LoadManager() {}
+
+  /**
+   * Validate the load information.
+   * @param loadInfo load information
+   * @return boolean value on whether the load is validated for scheduling or not
+   */
+  public boolean validate(LoadInfo loadInfo) {
+    LoadOptions options = loadInfo.getLoadOptions();
+    String path = loadInfo.getPath();
+    ValidationStatus status = getValidationStatus(path, options);
+
+    if (status.equals(ValidationStatus.New)) {
```

Comment:
You won't need the enum, but in the future, if you do, use `==` for enum equality.